app/src/androidTest/java/com/github/polypoly/app/network/RemoteDBTest.kt

```diff
@@ -16,27 +17,31 @@ import java.util.*
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.ExecutionException
 import java.util.concurrent.TimeUnit
+import kotlin.NoSuchElementException
 import kotlin.reflect.KClass
 
 @RunWith(JUnit4::class)
 class RemoteDBTest: PolyPolyTest(false, false) {
 
     private val rootTests = "test-hugo"
     var dbRootRef: DatabaseReference
+    private val testDB: RemoteDB
 
     init {
         val db = Firebase.database
         try {
             db.setPersistenceEnabled(false)
         } catch (_: Exception) {}
-        remoteDB = RemoteDB(db, rootTests)
+        testDB = RemoteDB(db, rootTests)
+        remoteDB = testDB
```

Comment:
why have both remoteDB and testDB fields?

Reply:
Maybe the names aren't the best. `remoteDB` is used in all the program, whereas `testDB` is instantiated locally. I made this to avoid test conflicts, before, some local DB tests (that use `MockDB`) put data in our real DB so with this assignation, I make sure that before each local/online DB test, the correct value is assigned to `remoteDB`